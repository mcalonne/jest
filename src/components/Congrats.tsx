
import { CongratsProps } from '../types/Guessed.app';
import { COMPONENT_DATA_TEST_ATT, DEFAULT_SUCCESS_MSG } from '../shared/Congrats.constants';

const Congrats: React.FC<CongratsProps> = ({ success, successMessage = DEFAULT_SUCCESS_MSG }) => {
    return success ?
        <div data-test={COMPONENT_DATA_TEST_ATT}>
            <span data-test='congrats-message'>
                {successMessage}
            </span>
        </div>
    : <div data-test={COMPONENT_DATA_TEST_ATT}/>
};

export default Congrats;