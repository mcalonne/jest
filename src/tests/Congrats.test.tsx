import { shallow, ShallowWrapper } from 'enzyme';
import renderer from 'react-test-renderer';
import { findTestAtt } from './testUtils';
import { COMPONENT_DATA_TEST_ATT } from '../shared/Congrats.constants';

import Congrats from '../components/Congrats';

const setup = (props = { success: false }): ShallowWrapper => {
    return shallow(<Congrats {...props} />);
}

describe('Congrats component', () => {
    test('renders the component without errors', () => {
        const wrapper = setup();
        const component = findTestAtt(wrapper, COMPONENT_DATA_TEST_ATT);
        expect(component.length).toBe(1);
    });

    it('matches snapshot', () => {
        const componentTree = renderer.create(<Congrats success={false} />).toJSON();
        expect(componentTree).toMatchSnapshot();

    });
    
    test('renders an empty text when the success prop is false', () => {
        const wrapper = setup();
        const component = findTestAtt(wrapper, COMPONENT_DATA_TEST_ATT);
        expect(component.text()).toBe('');
    });
    
    test('renders the successfull message when the success prop is true', () => {
        const wrapper = setup({ success: true });
        const component = findTestAtt(wrapper, COMPONENT_DATA_TEST_ATT);
        expect(component.text()).toBe('Congratulations! You guessed the world');
    });
});
