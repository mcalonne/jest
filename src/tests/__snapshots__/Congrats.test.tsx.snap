// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Congrats component matches snapshot 1`] = `
<div
  data-test="congrats-component"
/>
`;
